src/mscorlib/shared/System/SpanHelpers.Char.cs

```diff
@@ -167,6 +168,91 @@ public static unsafe int IndexOf(ref char searchSpace, char value, int length)
             }
         }
 
+        public static unsafe int LastIndexOf(ref char firstChar, int startIndex, char value, int length)
```

Comment:
Why do we need the startIndex argument? Isn't it always length - 1?

Reply:
I see. I think this should be length, just like for IndexOf.

Reply:
I don't see how that would work. 

Let's say we change the caller to pass in length for startIndex instead:
```C#
public int LastIndexOf(char value) => SpanHelpers.LastIndexOf(ref _firstChar, Length, value, Length);
```

For a string of length 1, we would end up with the incorrect result (-1 instead of 0).
```C#
'a'.LastIndexOf('a'); // Expected: 0, Actual: -1
```

We have to pass in Length - 1 as the startIndex since the character at position startIndex is included in the search. Which means, we have to special case Length == 0, as well.

Reply:
`IndexOf` and `LastIndexOf` implementation should be pretty similar. IndexOf helper takes 3 arguments, so LastIndexOf helper should take 3 arguments as well.

The arguments of the `SpanHelpers.LastIndexOf` method should be the block of memory (ie pointer + length) and the value you are looking for. The method should always start searching at `length - 1`. 

The naive non-vectorized algorithm would be:

```csharp
static int LastIndexOf(ref char searchSpace, char value, int length)
{
    for (int i = length - 1; i >= 0; i--)
        if (Unsafe.Add(ref searchSpace, i) == value) return i
    return -1;
}
```

And the call from `int IndexOf(char value, int startIndex, int count)` should be:

```csharp
static int LastIndexOf(char value, int startIndex, int count)
{
...
    int endIndex = startIndex + 1 - count;
    int result = SpanHelpers.LastIndexOf(ref Unsafe.Add(_firstChar, endIndex), count, value);
    return result == -1 ? result : result + endIndex;
```

Reply:
Got it. Thanks. That is what I ended up doing :)